src/Sylius/Bundle/ApiBundle/Validator/Constraints/ChoosePaymentMethodEligibilityValidator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\Constraints;
+
+use Sylius\Bundle\ApiBundle\Command\PaymentMethodCodeAwareInterface;
+use Sylius\Component\Core\Repository\PaymentMethodRepositoryInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+class ChoosePaymentMethodEligibilityValidator extends ConstraintValidator
+{
+    /** @var PaymentMethodRepositoryInterface */
+    private $paymentMethodRepository;
+
+    public function __construct(PaymentMethodRepositoryInterface $paymentMethodRepository)
+    {
+        $this->paymentMethodRepository = $paymentMethodRepository;
+    }
+    public function validate($value, Constraint $constraint): void
+    {
+        Assert::isInstanceOf($value, PaymentMethodCodeAwareInterface::class);
+
+        /** @var ChoosePaymentMethodEligibility $constraint */
+        Assert::isInstanceOf($constraint, ChoosePaymentMethodEligibility::class);
+
+        $paymentMethod = $this->paymentMethodRepository->findOneBy(['code' => $value->getPaymentMethodCode()]);
+
+        if ($paymentMethod === null) {
+            $this->context->addViolation(
+                $constraint->paymentMethodNotExistMessage,
+                ['%paymentMethodCode%' => $value->paymentMethodCode]
+            );
+
+            return;
```

Comment:
Do we need this `return`? Nothing is happening after `if` statement

Reply:
Just a little jump up the front, if we will extend this validator (and it will be happening for sure in nearest future) than this return will be valid 😉

Reply:
Then we will add it 😄 adding something even for the predictable future is not a best practice, as this future **can always change** (as it's in the future, so is uncertain 💃)